core/liquidity/v2/types.go

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) 2022 Gobalsky Labs Limited
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE.VEGA file and at https://www.mariadb.com/bsl11.
+//
+// Change Date: 18 months from the later of the date of the first publicly
+// available Distribution of this version of the repository, and 25 June 2022.
+//
+// On the date above, in accordance with the Business Source License, use
+// of this software will be governed by version 3 or later of the GNU General
+// Public License.
+
+package liquidity
+
+import (
+	"sort"
+
+	"code.vegaprotocol.io/vega/core/types"
+	"code.vegaprotocol.io/vega/libs/num"
+)
+
+// Provisions provides convenience functions to a slice of *vega/proto.LiquidityProvision.
+type Provisions []*types.LiquidityProvision
+
+// feeForTarget returns the right fee given a group of sorted (by ascending fee) LiquidityProvisions.
+// To find the right fee we need to find smallest index k such that:
+// [target stake] < sum from i=1 to k of [MM-stake-i]. In other words we want in this
+// ordered list to find the liquidity providers that supply the liquidity
+// that's required. If no such k exists we set k=N.
+func (l Provisions) feeForTarget(t *num.Uint) num.Decimal {
+	if len(l) == 0 {
+		return num.DecimalZero()
+	}
+
+	n := num.UintZero()
+	for _, i := range l {
+		n.AddSum(i.CommitmentAmount)
+		if n.GTE(t) {
+			return i.Fee
+		}
+	}
+
+	// return the last one
+	return l[len(l)-1].Fee
+}
+
+type lpsByFee Provisions
+
+func (l lpsByFee) Len() int           { return len(l) }
+func (l lpsByFee) Less(i, j int) bool { return l[i].Fee.LessThan(l[j].Fee) }
+func (l lpsByFee) Swap(i, j int)      { l[i], l[j] = l[j], l[i] }
+
+// sortByFee sorts in-place and returns the LiquidityProvisions for convenience.
+func (l Provisions) sortByFee() Provisions {
+	byFee := lpsByFee(l)
+	sort.Sort(byFee)
+	return Provisions(byFee)
+}
```

Comment:
Why not just do  this:
```suggestion
// sortByFee sorts in-place and returns the LiquidityProvisions for convenience.
func (l Provisions) sortByFee() Provisions {
	sort.Slice(l, func(i, j int) bool {return l[i].Fee.LessTna(l[j].Fee)})
	return Provisions(byFee)
}
```

Reply:
Because it was a CP and Paste and I have not cleared that file yet mate.